docs/ARCHITECTURE.md

```
@@ -0,0 +1,162 @@
+
+- [Clean Architecture Layers](#clean-architecture-layers)
+- [Dependency Rule](#dependency-rule)
+- [Dependency Diagram](#dependency-diagram)
+- [Module Groups](#module-groups)
+  - [Domain](#domain)
+  - [Data](#data)
+  - [Framework](#framework)
+  - [Presentation](#presentation)
+  - [Shared Infrastructure](#shared-infrastructure)
+- [design-system, ui, and feature:* Boundaries](#design-system-ui-and-feature-boundaries)
+- [Further Reading](#further-reading)
```

markdownlint-cli2 (0.23.2)

[warning] Link fragments should be valid

(MD051, link-fragments)